test/it/user-role.js

```diff
@@ -0,0 +1,147 @@
+  orgUrl: orgUrl,
+  token: process.env.OKTA_CLIENT_TOKEN,
+  requestExecutor: new okta.DefaultRequestExecutor()
+});
+
+describe('User role API', () => {
+  let user;
+  beforeEach(async () => {
+    user = await client.createUser(mockUser, { activate: false });
+  });
+  afterEach(async () => {
+    await utils.cleanupUser(client, user);
+  });
+
+  describe('Role assignment', () => {
+    let role;
+    afterEach(async () => {
+      user.removeRole(role.id);
+    });
+
+    it('should assign role to user', async () => {
+      role = await user.assignRole({ type: 'APP_ADMIN' });
+      expect(role).to.be.instanceOf(models.Role);
+      expect(role.id).to.be.exist;
+      expect(role.type).to.equal('APP_ADMIN');
+    });
+  });
+
+  describe('Role unassignment', () => {
+    let role;
+    beforeEach(async () => {
+      role = await user.assignRole({ type: 'APP_ADMIN' });
+    });
+
+    it('should unassign role from user', async () => {
+      const res = await user.removeRole(role.id);
+      expect(res.status).to.equal(204);
+    });
+  });
+
+  describe('List user assigned roles', () => {
+    let role;
+    beforeEach(async () => {
+      role = await user.assignRole({ type: 'APP_ADMIN' });
+    });
+    afterEach(async () => {
+      user.removeRole(role.id);
+    });
+
+    it('should return a Collection of roles', async () => {
+      const roles = await user.listRoles();
+      expect(roles).to.be.instanceOf(Collection);
+      await roles.each(roleFromCollection => {
+        expect(roleFromCollection).to.be.instanceOf(models.Role);
+        expect(roleFromCollection.id).to.be.equal(role.id);
+      });
+    });
+  });
+
+  describe('App targets for admin role', () => {
+    let role;
+    let application;
+    beforeEach(async () => {
+      role = await user.assignRole({ type: 'APP_ADMIN' });
+      const mockApplication = utils.getBookmarkApplication();
+      application = await client.createApplication(mockApplication);
+    });
+    afterEach(async () => {
+      await application.deactivate();
+      await application.delete();
+      await user.removeRole(role.id);
+    });
+
+    describe('Add app target', () => {
+      it('should add app target to admin user', async () => {
+        const res = await role.addAppTargetToAdminRoleForUser(user.id, application.name);
+        expect(res.status).to.equal(204);
+      });
+    });
+
+    describe('List app targets', () => {
+      beforeEach(async () => {
+        await role.addAppTargetToAdminRoleForUser(user.id, application.name);
+      });
+
+      it('should return a Collection of Applications', async () => {
+        const apps = await client.listApplicationTargetsForApplicationAdministratorRoleForUser(user.id, role.id);
+        expect(apps).to.be.instanceOf(Collection);
+        await apps.each(app => {
+          // TODO: right now app model is not resolvable due to "signOnModes" field in response cannot be handled in ApplicationFactory
```

Comment:
@robertjd My idea is change `getResolutionProperty` method to `getResolutionProperties` in `resolution-factory.js`. Then resolve models based on first found resolution property. Let me know if you have any questions before I move forward.

Reply:
Do we have a secondary discriminator in the openapi spec? I'd like to know more about this idea (but not in this PR)